addons/hr_recruitment/models/hr_recruitment.py

```diff
@@ -457,6 +458,10 @@ def create_employee_from_applicant(self):
                     'work_phone': applicant.department_id and applicant.department_id.company_id
                             and applicant.department_id.company_id.phone or False})
                 applicant.write({'emp_id': employee.id})
+                applicant.message_post_with_view(
+                    'hr_recruitment.applicant_hired_template',
+                    values={'applicant': applicant},
+                    subtype_id=self.env.ref("hr_recruitment.mt_applicant_hired").id)
```

Comment:
subtype is not a valid argument, leading to a warning on runbot.

The correct way to handle this is

subtype_id=self.env.ref("hr_recruitment.mt_applicant_hired").id
message_post_with_view has not the same signature than message_post by the way.